test/Runtime/demangleToMetadata.swift

```diff
@@ -78,12 +78,23 @@ DemangleToMetadataTests.test("metatype types") {
 
 func f2_any_anyobject(_: Any, _: AnyObject) { }
 
+class C { }
+
+protocol P1 { }
+protocol P2 { }
+
+func f1_composition(_: P1 & P2) { }
+func f1_composition_superclass(_: C & P1 & P2) { }
+
 DemangleToMetadataTests.test("existential types") {
   // Any, AnyObject
   expectEqual(type(of: f2_any_anyobject), _typeByMangledName("yyyp_yXltc")!)
 
-  // FIXME: References to protocols.
-  // FIXME: References to superclass.
+  // References to protocols.
```

Comment:
Just for completeness, can you test `P & AnyObject` where P is a non-class-bound protocol?

Reply:
Will do!